File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/AssetsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AssetsRow } from "./AssetsRow";
import { AssetType } from "./AssetType";

export interface AssetsColumns {
    AssetId: Column<AssetsRow>;
    AssetName: Column<AssetsRow>;
    SerialNumber: Column<AssetsRow>;
    AssetType: Column<AssetsRow>;
    Status: Column<AssetsRow>;
    AssignedToFullName: Column<AssetsRow>;
    PurchaseDate: Column<AssetsRow>;
    Cost: Column<AssetsRow>;
}

export class AssetsColumns extends ColumnsBase<AssetsRow> {
    static readonly columnsKey = 'Operations.Assets';
    static readonly Fields = fieldsProxy<AssetsColumns>();
}

[AssetType]; // referenced types

File: HRMS/HRMS.Web/Modules/Operations/Assets/AssetsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.Assets")]
[BasedOnRow(typeof(AssetsRow), CheckNames = true)]
public class AssetsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int AssetId { get; set; }
    [EditLink]
    public string AssetName { get; set; }
    public string SerialNumber { get; set; }
    public AssetType AssetType { get; set; }
    public AssetStatus Status { get; set; }
    public string AssignedToFullName { get; set; }
    public DateTime PurchaseDate { get; set; }
    public decimal Cost { get; set; }
}
